src/Sarif.Converters/SemmleConverter.cs

```diff
@@ -11,8 +15,189 @@ namespace Microsoft.CodeAnalysis.Sarif.Converters
     /// </summary>
     internal class SemmleConverter : ToolFileConverterBase
     {
+        // Semmle logs are CSV files.
+        private static readonly string[] s_delimiters = new[] { "," };
+
+        // The fields are as follows:
+        private enum FieldIndex
+        {
+            QueryName,
+            QueryDescription,
+            Severity,
+            Message,
+            RelativePath,
+            Path,
+            StartLine,
+            StartColumn,
+            EndLine,
+            EndColumn
+        }
+
+        private TextFieldParser _parser;
+        private List<Notification> _toolNotifications;
+
+        /// <summary>
+        /// Converts a Semmle log file in CSV format to a SARIF log file.
+        /// </summary>
+        /// <param name="input">
+        /// Input stream from which to read the Semmle log.
+        /// </param>
+        /// <param name="output">
+        /// Output string to which to write the SARIF log.
+        /// </param>
+        /// <exception cref="ArgumentNullException">
+        /// Thrown when one or more required arguments are null.
+        /// </exception>
         public override void Convert(Stream input, IResultLogWriter output)
         {
+            if (input == null)
+            {
+                throw new ArgumentNullException(nameof(input));
+            }
+
+            if (output == null)
+            {
+                throw new ArgumentNullException(nameof(output));
+            }
+
+            _toolNotifications = new List<Notification>();
+
+            var results = GetResultsFromStream(input);
+
+            var tool = new Tool
```

Comment:
>tool [](start = 16, length = 4)

I see that we don't have any other tool data we can extract from the output file. Too bad. Can we confirm the tool name with Semmle? I think they actually advertise this functionality as 'Semmle QL' in their product portfolio. Semmle itself being the company name.